src/GeniusYield/Transaction.hs

```diff
@@ -118,12 +118,15 @@ data BuildTxException
     | BuildTxBodyErrorAutoBalance !Api.TxBodyErrorAutoBalance
     | BuildTxMissingMaxExUnitsParam
     -- ^ Missing max ex units in protocol params
-    | BuildTxExUnitsTooBig
-    -- ^ Execution units required is higher than the maximum as specified by protocol params.
+    | BuildTxExUnitsTooBig  -- ^ Execution units required is higher than the maximum as specified by protocol params.
+        (Natural, Natural)  -- ^ Tuple of maximum execution steps & memory as given by protocol parameters.
+        (Natural, Natural)  -- ^ Tuple of execution steps & memory as taken by built transaction.
+
     | BuildTxSizeTooBig
     -- ^ Transaction size is higher than the maximum as specified by protocol params.
```

Comment:
@sourabhxyz  also add information about  maximum tx Size given by protocol parameters &  used by transaction?

Reply:
Done in later commit.